Emby.Server.Implementations/Library/DefaultAuthenticationProvider.cs

```diff
@@ -19,31 +20,111 @@ public DefaultAuthenticationProvider(ICryptoProvider crypto)
+            bool success = false;
+            if (resolvedUser == null)
+            {
+                success = false;
+                throw new Exception("Invalid username or password");
+            }
+            ConvertPasswordFormat(resolvedUser);
+            byte[] passwordbytes = Encoding.UTF8.GetBytes(password);
+           
+            if (!resolvedUser.Password.Contains("$"))
+            {
+                ConvertPasswordFormat(resolvedUser);
+            }
+            PasswordHash ReadyHash = new PasswordHash(resolvedUser.Password);
+            byte[] CalculatedHash;
+            string CalculatedHashString;
+            if (_cryptographyProvider.GetSupportedHashMethods().Any(i => i == ReadyHash.Id))
+            {
+                if (String.IsNullOrEmpty(ReadyHash.Salt))
+                {
+                    CalculatedHash = _cryptographyProvider.ComputeHash(ReadyHash.Id, passwordbytes);
+                    CalculatedHashString = BitConverter.ToString(CalculatedHash).Replace("-", string.Empty);
+                }
+                else
+                {
+                    CalculatedHash = _cryptographyProvider.ComputeHash(ReadyHash.Id, passwordbytes, ReadyHash.SaltBytes);
+                    CalculatedHashString = BitConverter.ToString(CalculatedHash).Replace("-", string.Empty);
+                }
+                if (CalculatedHashString == ReadyHash.Hash)
+                {
+                    success = true;
+                    //throw new Exception("Invalid username or password");
+                }
+            }
+            else
+            {
+                success = false;
+                throw new Exception(String.Format("Requested crypto method not available in provider: {0}", ReadyHash.Id));
+            }
+
+            //var success = string.Equals(GetPasswordHash(resolvedUser), GetHashedString(resolvedUser, password), StringComparison.OrdinalIgnoreCase);
+
+            if (!success)
+            {
+                throw new Exception("Invalid username or password");
+            }
+
+            return Task.FromResult(new ProviderAuthenticationResult
+            {
+                Username = username
+            });
+        }
 
-            return Task.FromResult(new ProviderAuthenticationResult
-            {
-                Username = username
-            });
+        //This allows us to move passwords forward to the newformat without breaking. They are still insecure, unsalted, and dumb before a password change
+        //but at least they are in the new format.
+        private void ConvertPasswordFormat(User user)
+        {
+            if (!string.IsNullOrEmpty(user.Password))
+            {
+                if (!user.Password.Contains("$"))
+                {
+                    string hash = user.Password;
+                    user.Password = String.Format("$SHA1${0}", hash);
+                }
+                if (user.EasyPassword != null && !user.EasyPassword.Contains("$"))

```

Comment:
Styling dictates there should be a line break after `if` if it isn't followed by `else if` or `else`
```suggestion
                
                if (user.EasyPassword != null && !user.EasyPassword.Contains("$"))
```